db/src/tx.rs

```diff
@@ -677,6 +689,12 @@ impl<'conn, 'a, W> Tx<'conn, 'a, W> where W: TransactWatcher {
                         (true, true) => fts_many.push(reduced),
                     }
                 },
+
+                Term::RetractEntity(KnownEntid(e)) => {
+                    // TODO: Might update metadata?
+                    // TODO: Watching/counting datoms?
```

Comment:
Yep, the observer service needs to be extended to be told about retracted entities.